### DISCUSSION TITLE
duplicate symbols for architecture x86_64 iOS build error on Apple Sillicon

### DISCUSSION THREAD

**Discussion by fandreirebaseit** [WalletConnect/walletconnect-monorepo#3221] [Developer Support] [answered]
Hi,

On apple sillicon arch, I have managed to build the project on Android after installing all the related walletconnect modal packages, but when building on iOS at the finish of the building I get 144 duplicate symbols for architecture x86_64 and also I see some duplicated .h files but can't acces them or delete the duplicates, what some of those folders have in common is Async and SSL (i think maybe from async storage package), they kinda appear just on error, like in the image below.

What I have tried till now:
- include x86_64 to excluded architectures
- in build settings -> OTHER_LDFLAGS remove "-ObjC"
- put use_flipper!({'Flipper-Folly' => '2.6.7', "Flipper-DoubleConversion" => "1.1.7"}) in podfile but I have an error:
  CocoaPods could not find compatible versions for pod "Flipper-DoubleConversion":
  In Podfile:
    Flipper-DoubleConversion (= 1.1.7)
    Flipper-DoubleConversion (= 3.1.7)
There are duplicate dependencies on `Flipper-DoubleConversion` in `Podfile`:
   Flipper-DoubleConversion (= 1.1.7)
   Flipper-DoubleConversion (= 3.1.7)

- tried to change podfile to use use_framework() instead of use_flipper(), it installs DoubleConversion => 1.1.7 but doesn't install all the other pods and have error on every other pods when building 

Do you have any ideas on how to solve that ? 

PS: IMPORTANT! I have not deleted the older wallet connect v1 packages cause there are a lot of dependencies in the code all over the project and I wanted to be sure that is running on v2 with the setup

<img width="357" alt="Screenshot 2023-07-19 at 17 59 52" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/109270089/ab0f57fe-c0e8-47a9-94f8-4ba21d7a4047">

**Comment by ignaciosantise** [comment]
hey @fandreirebaseit 👋 Those packages you mention are not related with our SDK, but with react-native itself. Did you recently updated your project's react-native version? Maybe there was a problem with the update. Did you try removing the `Pod` folder and installing them again with `pod install`?

**Reply by fandreirebaseit** [reply]
hey, I have updated react-native version like 2 months ago to 0.65 from 0.64 but the v1 was running on 0.65 updated version, yeah I have tried to remove Pods folder and pod install after

**Reply by ignaciosantise** [reply]
try also removing the Podfile.lock before reinstalling pods

**Reply by fandreirebaseit** [reply]
have done that already

**Comment by fandreirebaseit** [comment, marked as answer]
I have managed to solved it, the problem was that there were duplicated files generated by react-native-udp and CocoaAsyncSocket. I needed to go in xcode to Pods -> TcpSockets pod -> Build phases -> Compile sources and delete the GCDAsyncSocket which is from CocoaAsyncSocket folder in node_modules and before need to make sure that react-native-udp is up to date in the project cause it was in an older version for my current rn version
